### PR TITLE
Fix ecl access cells

### PR DESCRIPTION
**Pull request by bergbauer**
References https://github.com/dealii/dealii/pull/15959#issuecomment-1876559412

@peterrum

### CODE REVIEW

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -8468,8 +8468,23 @@ FEFaceEvaluation<dim,
     {
       this->face_orientations[0] = 0;
       this->face_numbers[0]      = face_number;
-      for (unsigned int i = 0; i < n_lanes; ++i)
-        this->cell_ids[i] = cell_index * n_lanes + i;
+      if (this->matrix_free->n_active_entries_per_cell_batch(this->cell) ==
+          n_lanes)
+        {
+          DEAL_II_OPENMP_SIMD_PRAGMA
```
Comment (peterrum) [inline include/deal.II/matrix_free/fe_evaluation.h:8474]: Do we need this?

**Review by peterrum — APPROVED** [review]
/rebuild

File: include/deal.II/matrix_free/fe_evaluation.h (hunk 1 above)
Comment (bergbauer) [reply include/deal.II/matrix_free/fe_evaluation.h:8474]: I have copied it from here https://github.com/dealii/dealii/blob/c03f51671b0d04ede557bf9a0a1a893c470d5150/include/deal.II/matrix_free/fe_evaluation.h#L7488-L7502

**Review by kronbichler — APPROVED** [review]
OK with me. I would probably have written first a loop over all `n_lanes` unconditionally and then a second loop
```c++
for (unsigned int i = this->matrix_free->n_active_entries_per_cell_batch(this->cell);
       i < n_lanes; ++i)
```
but that is a matter of taste.

**Pull request merged**